### PR TITLE
Virial interface for AtomsCalculator

### PR DESCRIPTION
**Pull request by CedricTravelletti**
This PR adds virial calculation to the DFTK AtomsCalculators inteface.

It also fixes the units in the AtomsCalculators interface. 

Indeed, AtomsCalculators.jl specifies that we should return Unitful results.

### CODE REVIEW

File: src/external/atoms_calculators.jl [hunk 1]
```diff
@@ -69,12 +71,20 @@ AtomsCalculators.@generate_interface function AtomsCalculators.potential_energy(
...
-    compute_forces_cart(calculator.state.scfres)
+    compute_forces_cart(calculator.state.scfres) * u"hartree/bohr"
+end
+
+AtomsCalculators.@generate_interface function AtomsCalculators.virial(
+        system::AbstractSystem, calculator::DFTKCalculator; state = DFTKState(),
+        kwargs...)
+    compute_scf!(system, calculator, state)
+    - (compute_stresses_cart(calculator.state.scfres)
+     * calculator. state.scfres.basis.model.unit_cell_volume) * u"hartree"
```
Comment (mfherbst) [inline src/external/atoms_calculators.jl:89]: There is a space too much after the `calculator.`

Plus are you sure this is the right unit ? I had in mind it was something of the dimensions energy / volume, but I could be wrong.

File: test/external/atoms_calculators.jl [hunk 2]
```diff
@@ -15,10 +17,52 @@
 
     energy = AtomsCalculators.potential_energy(silicon, calculator)
     ref_energy = -7.86059
-    @test energy ≈ ref_energy rtol=1e-3
+    @test austrip.(energy) ≈ ref_energy rtol=1e-3
```
Comment (mfherbst) [inline test/external/atoms_calculators.jl:20]: no `austrip.`, just `austrip` (this is a scalar, no vector)

File: src/external/atoms_calculators.jl (hunk 1 above)
Comment (CedricTravelletti) [reply src/external/atoms_calculators.jl:89]: According to [AtomsCalculators](https://juliamolsim.github.io/AtomsCalculators.jl/dev/interface-definition/) interface definition: 
`Virial is a square matrix (3x3 in 3D) that has units of force times length or energy`, so here hartrees are appropriate.

**Comment by CedricTravelletti** [comment]
Thank your for the review. 
Comments have been adressed. Ready for another round of review.

File: test/external/atoms_calculators.jl [hunk 3]
```diff
@@ -15,10 +17,52 @@
...
+    silicon = TestCases.silicon
+
+    # Perturb from equilibrium so forces are not 0.
+    positions = [silicon.positions[1] + [0.05, 0, 0], silicon.positions[2]]
+    silicon = periodic_system(silicon.lattice, silicon.atoms, positions)
+
+    model_kwargs = (; temperature=1e-3, functionals=[:lda_x, :lda_c_pw])
+    basis_kwargs = (; kgrid=[4, 4, 4], Ecut=5.0)
+    scf_kwargs = (; tol=1e-7)
+    calculator = DFTKCalculator(; model_kwargs, basis_kwargs, scf_kwargs, verbose=true)
```
Comment (mfherbst) [inline test/external/atoms_calculators.jl:38]: This is now essentially 3 times the same code. Can you make a function singling this out (see other files how this is done).

File: src/external/atoms_calculators.jl [hunk 4]
```diff
@@ -69,12 +71,20 @@ AtomsCalculators.@generate_interface function AtomsCalculators.potential_energy(
...
 AtomsCalculators.@generate_interface function AtomsCalculators.forces(
         system::AbstractSystem, calculator::DFTKCalculator; state = DFTKState(),
         kwargs...)
     compute_scf!(system, calculator, state)
-    compute_forces_cart(calculator.state.scfres)
+    compute_forces_cart(calculator.state.scfres) * u"hartree/bohr"
+end
+
+AtomsCalculators.@generate_interface function AtomsCalculators.virial(
+        system::AbstractSystem, calculator::DFTKCalculator; state = DFTKState(),
```
Comment (mfherbst) [inline src/external/atoms_calculators.jl:85]: We don't put spaces around `=` in Keyword arguments (same above, actually, which I did not catch last time)

**Review by mfherbst — COMMENTED** [review]
Rest looks good to me.

File: test/external/atoms_calculators.jl (hunk 3 above)
Comment (CedricTravelletti) [reply test/external/atoms_calculators.jl]: fixed

**Comment by CedricTravelletti** [comment]
@mfherbst Do you think there are other modifications to be done or we can merge?

**Pull request merged**